### DISCUSSION TITLE
Long Delay For Action

### DISCUSSION THREAD

**Discussion by eboston** [HASwitchPlate/HASPone#8] [Q&A] [answered]
I got the new version added to a new unit and so far has been pretty easy getting things going.  However, I am seeing long delays after pressing a button before the action is performed.  Sometime, the action is instant, but other times there could be a delay of 10 seconds or more after the button is pressed.  Is this to be expected or is there something else that can be done to reduce the delay?

**Comment by aderusha** [comment]
If you use an MQTT logging tool (MQTT Explorer or similar), can you see if the delay is between you pressing the button and HASP sending the message, or HASP sending the message and Home Assistant responding?

**Reply by eboston** [reply]
I am seeing changes in MQTT Explorer right when I touch a button.  It looks like HASP is sending the message right away but HA is not processing it immediately.

**Reply by eboston** [reply]
I might have to take that back.  The HASP seems to be having network issues.  I see it jumping back to page 0 with the network info and then back to the page it was on.

**Comment by aderusha** [comment]
One thing I discovered - disabling the packages for the 6 HASPs I’m running resulted in a huge increase in overall performance on my main Home Assistant install.  I guess having nearly 1000 automations has that effect...

**Comment by eboston** [comment, marked as answer]
Problem is with the UniFi APs that I have.  Added a non-UniFi AP and when connected to that, button press response is as expected.
